File: contradictionDetection.py
from transformers import pipeline

# Load a pre-trained NLI model from Hugging Face
nli_model = pipeline("text-classification", model="facebook/bart-large-mnli")

def check_contradiction(premise, hypothesis):
    # Format the input correctly for the NLI model
    inputs = f"{premise} [SEP] {hypothesis}"
    
    # Use the NLI model to classify the relationship between premise and hypothesis
    result = nli_model(inputs)
    
    # Extract the label (entailment, contradiction, or neutral)
    label = result[0]['label']
    score = result[0]['score']
    
    return label, score

# Example usage
# generated_text = "The Eiffel Tower is located in New York."
generated_text = "The Eiffel Tower is in Tianducheng, China."
ground_truth = "The Eiffel Tower is located in Paris."

# Check for contradiction
label, confidence = check_contradiction(generated_text, ground_truth)

print(f"Label: {label}, Confidence: {confidence}")

File: typescript/contradictionDetection.js
const huggingface = require("@huggingface/inference");

// Hugging Face API details
const API_URL =
  "https://api-inference.huggingface.co/models/facebook/bart-large-mnli";
const API_TOKEN = "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";

// Function to detect contradiction
async function detectContradiction() {
  const inference = new huggingface.HfInference(
    "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"
  );
  const model = "facebook/bart-large-mnli";

  const result = await inference.zeroShotClassification({
    model: model,
    inputs: "I like your outfit!",
    parameters: {
      // candidate_labels: ["contradiction", "entailment", "neutral"],
      candidate_labels: ["insult", "friendly", "neutral"],
    },
  });

  console.log(result);
}

// Example usage
detectContradiction();
